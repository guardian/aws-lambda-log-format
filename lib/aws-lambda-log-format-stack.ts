import { Stack, StackProps, Tags } from "aws-cdk-lib";
import { Construct } from "constructs";
import { NodejsFunction } from "aws-cdk-lib/aws-lambda-nodejs";
import { LoggingFormat, Runtime } from "aws-cdk-lib/aws-lambda";
import * as path from "node:path";
import { NodejsFunctionProps } from "aws-cdk-lib/aws-lambda-nodejs/lib/function";

export class AwsLambdaLogFormatStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    Tags.of(this).add("Stack", "playground");
    Tags.of(this).add("Stage", "TEST");
    Tags.of(this).add("gu:repo", "guardian/aws-lambda-log-format");

    const lambdaProps: NodejsFunctionProps = {
      runtime: Runtime.NODEJS_20_X,
      handler: "main",
      entry: path.join(__dirname, `/../src/lambda.ts`),
      bundling: {
        minify: true,
        externalModules: ["@aws-sdk"],
      },
      environment: {
        STACK: "playground",
        STAGE: "TEST",
      }
    }

    const textFormatFunction = new NodejsFunction(this, "TextFormatFunction", {
      ...lambdaProps,
      environment: {
        ...lambdaProps.environment,
        APP: "text-format-function",
      },
      loggingFormat: LoggingFormat.TEXT,
    });

    Tags.of(textFormatFunction).add("App", "text-format-function");

    const jsonFormatFunction = new NodejsFunction(this, "JsonFormatFunction", {
      ...lambdaProps,
      environment: {
        ...lambdaProps.environment,
        APP: "json-format-function",
      },
      loggingFormat: LoggingFormat.JSON,
    });

    Tags.of(jsonFormatFunction).add("App", "json-format-function");
  }
}
