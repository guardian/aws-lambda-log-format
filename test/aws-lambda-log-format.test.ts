import { Template } from "aws-cdk-lib/assertions";
import { App } from "aws-cdk-lib";
import { AwsLambdaLogFormatStack } from "../lib/aws-lambda-log-format-stack";

describe("The AwsLambdaLogFormatStack stack", () => {
 it("matches the snapshot", () => {
  const app = new App();
  const stack = new AwsLambdaLogFormatStack(app, "ServiceCatalogue", {
   env: { region: "eu-west-1" },
  });
  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
 });
});
