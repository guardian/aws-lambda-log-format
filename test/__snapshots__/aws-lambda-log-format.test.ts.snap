// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AwsLambdaLogFormatStack stack matches the snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "JsonFormatFunction71AA185D": {
      "DependsOn": [
        "JsonFormatFunctionServiceRoleAD45E999",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-eu-west-1",
          },
          "S3Key": "428aa57d905cd16a0a91b7403879470a25e76947ffbeedb9182d9af8b596d1bd.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "json-format-function",
            "STACK": "playground",
            "STAGE": "TEST",
          },
        },
        "Handler": "index.main",
        "LoggingConfig": {
          "LogFormat": "JSON",
        },
        "Role": {
          "Fn::GetAtt": [
            "JsonFormatFunctionServiceRoleAD45E999",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "json-format-function",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/aws-lambda-log-format",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::Lambda::Function",
    },
    "JsonFormatFunctionServiceRoleAD45E999": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "json-format-function",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/aws-lambda-log-format",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "TextFormatFunctionD37B24FE": {
      "DependsOn": [
        "TextFormatFunctionServiceRole6E5B2423",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-eu-west-1",
          },
          "S3Key": "428aa57d905cd16a0a91b7403879470a25e76947ffbeedb9182d9af8b596d1bd.zip",
        },
        "Environment": {
          "Variables": {
            "APP": "text-format-function",
            "STACK": "playground",
            "STAGE": "TEST",
          },
        },
        "Handler": "index.main",
        "LoggingConfig": {
          "LogFormat": "Text",
        },
        "Role": {
          "Fn::GetAtt": [
            "TextFormatFunctionServiceRole6E5B2423",
            "Arn",
          ],
        },
        "Runtime": "nodejs20.x",
        "Tags": [
          {
            "Key": "App",
            "Value": "text-format-function",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/aws-lambda-log-format",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::Lambda::Function",
    },
    "TextFormatFunctionServiceRole6E5B2423": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
        "Tags": [
          {
            "Key": "App",
            "Value": "text-format-function",
          },
          {
            "Key": "gu:repo",
            "Value": "guardian/aws-lambda-log-format",
          },
          {
            "Key": "Stack",
            "Value": "playground",
          },
          {
            "Key": "Stage",
            "Value": "TEST",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
